packages/editor/src/components/rich-text/index.js

```diff
@@ -113,14 +111,19 @@ export class RichText extends Component {
 		this.state = {};
 
 		this.usedDeprecatedChildrenSource = Array.isArray( value );
+		this.lastHistoryValue = value;
 	}
 
 	componentDidMount() {
 		document.addEventListener( 'selectionchange', this.onSelectionChange );
+		window.addEventListener( 'mousedown', this.onCreateUndoLevel );
```

Comment:
Still. Why are we binding regardless of whether the `RichText` is selected?

Reply:
I still don't understand what a mouse gesture has to do with undo, nor agree with assumption that use of mouse implies some milestone in the user's writing flow. Was this inherited from TinyMCE?

Reply:
As written, this would be a good candidate for `withGlobalEvents`:

https://github.com/WordPress/gutenberg/tree/master/packages/compose/src/with-global-events

Reply:
Oh, I see. Sure we can bind on selection, unbind on deselect.

> I still don't understand what a mouse gesture has to do with undo, nor agree with assumption that use of mouse implies some milestone in the user's writing flow. Was this inherited from TinyMCE?

Yes, TinyMCE has the same logic, which I think makes sense. E.g. if you type something, then move to another place in the text and continue to write, the undo button should not undo all typing at once.